inspirehep/records/marshmallow/authors/base.py

```diff
@@ -55,6 +57,37 @@ def get_should_display_positions(data):
 
         return True
 
+    @staticmethod
+    def get_author_with_record_facet_author_name(author):
+        author_ids = author.get("ids", [])
+        author_bai = get_values_for_schema(author_ids, "INSPIRE BAI")
+        bai = author_bai[0] if author_bai else "BAI"
+        author_preferred_name = get_value(author, "name.preferred_name")
+
+        if author_preferred_name:
+            return "{}_{}".format(bai, author_preferred_name)
+        else:
+            return "{}_{}".format(
+                bai,
+                AuthorsMetadataSchemaV1.get_author_display_name(
+                    author["name"]["value"]
+                ),
+            )
+
+    @staticmethod
+    def get_author_display_name(name):
+        parsed_name = ParsedName.loads(name)
+        return " ".join(parsed_name.first_list + parsed_name.last_list)
+
+    def get_facet_author_name(self, data):
```

Comment:
this should be place before `get_author_with_record_facet_author_name` for readability.